rpcs3/Emu/Io/usio.cpp

```diff
@@ -9,6 +9,9 @@
 
 LOG_CHANNEL(usio_log);
 
+constexpr u8 DefaultLastGameStatus[40] = {0x4C, 0x41, 0x53, 0x54, 0x47, 0x41, 0x4D, 0x45, 0x53, 0x54, 0x41, 0x54, 0x55, 0x53, 0x20, 0x76, 0x65, 0x72, 0x2E, 0x33, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00}; // "LASTGAMESTATUS ver.3"
+std::vector<u8> LastGameStatus;
```

Comment:
This does not work after an emulation exit and relaunch, this is why we use g_fxo instead of static variables.

Reply:
Yeah, I know that the data won't be preserved after the emulation exit.
The "LastGameStatus" is not something like record saving; instead, it is for the game to keep track of the last game status, such as how many remaining credits there were before a player resets the game (not relaunch the emulation).

Reply:
Still, better for it to be constant at launch for debugging purposes.

Reply:
Could you advise what the code should be like? Many thanks in advance.

Reply:
You create a new structure wrapping the vector, and you use g_fxo->get<custom_struct>().game_status instead of LastGameStatus.

Reply:
> You create a new structure wrapping the vector, and you use g_fxo->get<custom_struct>().game_status instead of LastGameStatus.

fixed_typemap.hpp:136 : C2338 static_assert failed: 'Please make sure the object cannot be accidentally copied.'

Reply:
ah right you need delete the copy constructor and assignment operator.

Reply:
> ah right you need delete the copy constructor and assignment operator.

The struct is just:
`struct LastGameStatus
{
	std::vector<u8> memory;
};`

Reply:
```struct last_game_status_t
{
     std::vector<u8> memory;

     last_game_status_t(const last_game_status_t&) = delete;
     last_game_status_t& operator=(const last_game_status_t&) = delete;
};
```
Replace spaces with tabs in this.

Reply:
Updated. Thank you.